bridgesync/downloader.go

```diff
@@ -306,6 +312,29 @@ func buildRemoveLegacyTokenHandler(contract *bridgel2sovereignchain.Bridgel2sove
 	}
 }
 
+// buildUpdatedClaimedGlobalIndexHashChainHandler creates a handler for the UpdatedClaimedGlobalIndexHashChain event log.
+func buildUpdatedClaimedGlobalIndexHashChainHandler(contract *bridgel2sovereignchain.Bridgel2sovereignchain) func(*sync.EVMBlock,
+	types.Log) error {
+	return func(b *sync.EVMBlock, l types.Log) error {
+		event, err := contract.ParseUpdatedClaimedGlobalIndexHashChain(l)
+		if err != nil {
+			return fmt.Errorf("error parsing UpdatedClaimedGlobalIndexHashChain event log %+v: %w", l, err)
+		}
+
+		b.Events = append(b.Events, Event{UpdatedClaimedGlobalIndexHashChain: &UpdatedClaimedGlobalIndexHashChain{
```

Comment:
@vcastellm what is this event used for?

Reply:
In this PR we introduce this event to order the has chain correctly, before that it wasn't used in the system AFAIK

Reply:
i was thinking can we insert this in claims db only and then we wont need joins or anything if both of them are related to claims